src/serializers/config.rs

```diff
@@ -125,6 +126,14 @@ impl TimedeltaMode {
                 let seconds = Self::total_seconds(&py_timedelta)?;
                 Ok(seconds.into_py(py))
             }
+            Self::Millisecond => {
+                // convert to int via a py timedelta not duration since we know this this case the input would have
+                // been a py timedelta
+                let py_timedelta = either_delta.try_into_py(py)?;
+                let seconds: f64 = Self::total_seconds(&py_timedelta)?.extract()?;
```

Comment:
There might be a better way to do this which is alluding me, maybe we could do the multiplication in python? 🤷🏻‍♂️

Reply:
Seems reasonable enough - multiplication here should be faster.

Can we pull out some of the shared logic into a function (both for Float and for Millisecond) and repeat that across the various branches?

Reply:
Hm, maybe, a couple of them use logic like:

```
let py_timedelta = either_delta.try_into_py(py)?;
let seconds: f64 = Self::total_seconds(&py_timedelta)?.extract()?;
```

However then the serializer needs to wrap these in `map_err` calls, so can't be reused there..

Reply:
To do the multiplication in Python would be something like

```rust
let object = Self::total_seconds(&py_timedelta)?.mul(1000)?;
```

... this requires creating a Python integer `1000`, which for best performance we might want to consider caching.

But there is another inefficiency here (and in the other cases) which is that the call to `try_into_py` creates a Python `timedelta` object which then gets thrown away immediately to convert to float. It will probably be better in all cases to use `.to_duration()`, which will avoid the temporary Python object in the case of a `Duration` object.

The _best_ option would be to go further and to  add a `.total_seconds()` method to `EitherTimedelta` which extracts the fractional seconds from whatever state the `EitherTimedelta` is currently in (doing the most efficient thing for each case) and then doing the multiplication in Rust.

Reply:
hmm, yeah neat suggestion on the `EitherTimedelta`

I guess we'd want something like this:

```rust
impl<'a> EitherTimedelta<'a> {
    ....

    pub fn total_seconds(&self, py: Python<'a>) -> f64 {
        match self {
            Self::Raw(timedelta) => ...
            Self::PyExact(py_timedelta) => ...
            Self::PySubclass(py_timedelta) => ...
        }
    }
}
```

And then we have two cases: `Duration` and `PyDelta` to deal with. Looks like we have some methods around that would involve getting the `py_timedelta` into a `Duration` object, so then its just `Duration` we need to deal with. However (maybe im reading the docs wrong!) i can't seem to see a method on there that returns the total_seconds as a float?

Probably missed something here!

Reply:
```rust
    pub fn total_seconds(&self, py: Python<'a>) -> f64 {
        match self {
            Self::Raw(timedelta) => ...,
            Self::PyExact(py_timedelta) => intern!(py_timedelta.py(), "total_seconds"))?.extract?
            Self::PySubclass(py_timedelta) => intern!(py_timedelta.py(), "total_seconds"))?.extract?
        }
    }
```

Something like this? Not 100% sure what to do with the `Raw` case, but doing what we do currently and extracting from the python object by calling "total_seconds" on it _feels_ like the best way in this case.

Reply:
I suggest looking at the `to_duration` method to see how that gets days / seconds / microseconds out of the Python value, as that should have the most efficient implementation already set up to get the data out (and then you can do a bit of arithmetic). In general calling a Python method will be slow-ish, so there might be a faster step for `PyExact` case.

For `Raw`, you need to work with the `speedate::Duration` object, can probably get a timestamp value and microseconds and combine those.